packages/cli/src/deploy/logsToWorldDeploy.ts

```diff
@@ -0,0 +1,49 @@
+import { AbiEventSignatureNotFoundError, Log, decodeEventLog, hexToString, parseAbi, trim } from "viem";
+import { WorldDeploy, helloStoreEvent, helloWorldEvent } from "./common";
+import { isDefined } from "@latticexyz/common/utils";
+
+export function logsToWorldDeploy(logs: readonly Log<bigint, number, false>[]): Omit<WorldDeploy, "toBlock"> {
+  const deployLogs = logs
+    .map((log) => {
+      try {
+        return {
+          ...log,
+          ...decodeEventLog({
+            strict: true,
+            abi: parseAbi([helloWorldEvent, helloStoreEvent]),
+            topics: log.topics,
+            data: log.data,
+          }),
+        };
+      } catch (error: unknown) {
+        if (error instanceof AbiEventSignatureNotFoundError) {
+          return;
+        }
+        throw error;
+      }
+    })
+    .filter(isDefined);
+
+  // TODO: should this test for/validate that only one of each of these events is present? and that the address/block number don't change between each?
```

Comment:
Imo we can just assume that, since the verified contracts only emit it once, and if a user actively re-emits it, it's either their fault if it breaks, or maybe it's for some intentional advanced use case. (There are many other ways in which users could deliberately make things break in their own custom World that we don't validate either.)